architecture/transaction-layer.md

```diff
@@ -0,0 +1,204 @@
+
+In terms of transactions, the gateway node picks a timestamp for the transaction using HLC time. Whenever a transaction's timestamp is mentioned, it's an HLC value. This timestamp is used to both track versions of values (through multiversion concurrency control, or [MVCC](storage-layer.html#mvcc)), which is detailed in the Storage Layer overview), as well as provide our transactional isolation guarantees.
+
+When nodes send requests to other nodes, the sending node includes a timestamp generated by their local HLCs (which includes both physical and logical components). When nodes receive requests, they inform their local HLC of the timestamp supplied with the event by the sender. This is useful in guaranteeing that all data read/written on a node is at a timestamp less than the next HLC time.
+
+This then lets the node primarily responsible for the range (i.e., the Lease Holder) serve reads for data it stores by ensuring the transaction reading the data is at an HLC time greater than the MVCC value it's reading (i.e., the read always happens "after" the write).
+
+#### Max Clock Offset Enforcement
+
+To ensure correctness among distributed nodes, you can identify a Maximum Clock Offset. Because CockroachDB relies on clock synchronization, nodes run a version of [Marzullo’s algorithm](http://infolab.stanford.edu/pub/cstr/reports/csl/tr/83/247/CSL-TR-83-247.pdf) amongst themselves to measure maximum clock offset within the cluster. If the configured maximum offset is exceeded by any node, it commits suicide, preventing it from potentially creating consistency issues within the cluster.
+
+For more detail about the risks that large clock offsets can cause, see [What happens when node clocks are not properly synchronized?](operational-faqs.html#what-happens-when-node-clocks-are-not-properly-synchronized)
+
+#### Timestamps & Serializability
+
+HLC timestamps let us understand the sequence of state changes in the database. CockroachDB then uses this ordering to provide its isolation guarantees. Isolation, in this context, meaning the order in which keys and their values become accessible to other operations. By providing strict isolation levels, we aim to provide anomaly-free consistency to our users, which we refer to as "serializability"––the database's state is ordered in a strictly serialized way, clearly ordered by HLC timestamps.
+
+For instance, if an operation finds a value with a timestamp in the future, it "understands" that its own timestamp must be moved forward to preserve serialized order in the database.
+
+However, when timestamps are moved, it introduces anomalies into the database. This is an intuitive concept, that it violates the notion of strict serialiazability to reorder the state of your data's values.
+
+In this document, you can find more about our [isolation levels]().
```

Comment:
The link is missing here, and I'd rephrase as `For more about our isolations levels, see xxx`.